examples/adhoc.py

```diff
@@ -4,21 +4,32 @@
 
 sta1 <---> sta2 <---> sta3"""
 
-from mininet.net import Mininet
 from mininet.cli import CLI
 from mininet.log import setLogLevel
+from mininet.net import Mininet
+from mininet.wifiPropagationModels import powerForRangeByPropagationModel
+
 
 def topology():
     "Create a network."
     net = Mininet(enable_wmediumd=True, enable_interference=True)
 
     print "*** Creating nodes"
     sta1 = net.addStation('sta1', position='10,10,0')
-    sta2 = net.addStation('sta2', position='90,10,0')
-    sta3 = net.addStation('sta3', position='170,10,0')
+    sta2 = net.addStation('sta2', position='50,10,0')
+    sta3 = net.addStation('sta3', position='90,10,0')
+
 
     net.propagationModel("logDistancePropagationLossModel", exp=4)
 
+    # calc for sta1, the others should be the same
+    power_to_distance = powerForRangeByPropagationModel().logDistancePropagationLossModel(node=sta1, wlan=0,
+                                                                                          distance=50)
+
+    sta1.setTxPower(sta1.params['wlan'][0], power_to_distance)
+    sta2.setTxPower(sta2.params['wlan'][0], power_to_distance)
+    sta3.setTxPower(sta3.params['wlan'][0], power_to_distance)
+
```

Comment:
Eduardo,

What do you think about including the code above in net.configureWifiNodes()?

Reply:
Something like in here (https://github.com/intrig-unicamp/mininet-wifi/blob/master/mininet/wifiNet.py#L1047) ? 
The thing is, what if the user sets a TxPower and then when they configure the wifi nodes it gets overwritten? 

This way of usage, tells the user explicitly that for getting a particular range, they need to set the transmission power to that value.

Adding to this observation, the range parameter in the build of the station, should be deprecated and throw and exception, since it doesn't not reflect what a user would expect?

Reply:
Yes, it makes sense. You're right. What if we add something like autoTxPower=bool in net.mininet()? I think it will solve the problem. As a consequence, Adhoc.py should look like this:

def topology(autoTxPower):
net = Mininet(enable_wmediumd=True, enable_interference=True, autoTxPower=autoTxPower)

if __name__ == '__main__':
    setLogLevel('info')
    autoTxPower = True if '-a' in sys.argv else False
    topology(autoTxPower)

What do you think?

Reply:
It makes sense to make it that easy and mininet adjust automatically its parameters to the range provided.

But there should be a way to detect mistakes if the user have set the range and the TXPower. Having the user to explicitly set the TXPower adjusted to the range makes it obvious that if they want to have a particular range the TXPower value must be that. (Or they should go read the propagation model and adjust the antenna gains).

Another option could be on the node.setRange(...) to update the txpower, but then if the user changes the txpower or was already changed in the past there should be some indication that it may have wrong results.